modules/internal/DefaultRectangular.chpl

```diff
@@ -644,6 +644,13 @@ module DefaultRectangular {
     pragma "local field"
     var shiftedData : _ddata(eltType);
     var noinit_data: bool = false;
+
+    // 'dataAllocDom' is used by the array-vector operations (e.g. push_back,
+    // pop_back, insert, remove) to allow growing or shrinking the data
+    // buffer in a doubling/halving style.  If it is used, it will be the
+    // actual size of the data buffer, while 'dom' represents the size of the
+    // user-level array.
+    var dataAllocDom: domain(1);
```

Comment:
Do we have any good way to tighten this up in order to avoid adding a whole 1D domain field to every default rectangular array that we allocate, whether it's an array-as-vec case or not?  Seems a bit clumsy and heavyweight.  Would it be possible to simply use an integer or a range for this purpose?  Do we know statically whether or not this field is used for a given array such that, if we had working void types, we could optimize the field away completely in the non-array-as-vec cases?